packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2108,10 +2028,72 @@ abstract class TreeDiagnosticsMixin implements TreeDiagnostics {
   /// }
   /// ```
   ///
-  /// Used by [toStringDeep], [toDiagnosticsNode] and [toStringShallow].
+  /// Used by [toDiagnosticsNode] and [toString].
   @protected
   @mustCallSuper
   void debugFillProperties(List<DiagnosticsNode> properties) { }
+}
+
+/// A class that can be used as a mixin that helps dump string and
+/// [DiagnosticsNode] representations of trees.
+///
+/// Use this class as a mixin in any context where it is possible to use mixins.
```

Comment:
these docs say they're for a mixin but you have a very similar mixin class so presumably these are to be a base class?